ping.go

```diff
@@ -339,6 +340,46 @@ func (p *Pinger) Addr() string {
 	return p.addr
 }
 
+// SetSrcAddr set source by interface name or ipv4 address
+// Interface name can be "eth0" or ip like "192.168.1.239"
+func (p *Pinger) SetSrcAddr(iface string) error {
+
+	ipPattern := regexp.MustCompile(`(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}`)
+	if ipPattern.MatchString(iface) {
```

Comment:
This should probably use go [`net.ParseIP()`](https://golang.org/pkg/net/#ParseIP).

Reply:
Cool, I will replace it with net.ParseIP().